app/controllers/attendance_sign_in_remote_controller.rb

```diff
@@ -12,8 +12,8 @@ def new
   def create
     record = AttendanceRecord.find_or_initialize_by(student: current_student, date: Time.zone.now.in_time_zone(current_student.course.office.time_zone).to_date)
     pair_ids = params[:pair_ids].uniq.reject {|p| p.empty?}
-    record.pair_ids = pair_ids
-    if record.save
+    record.pairings.delete_all
```

Comment:
Why is this in the controller?

Reply:
updating with dependent attributes actually adds them rather than replaces them... which is why deleting existing pairings is needed. I like the idea of moving that to AttendanceRecord#update_pairings or something like that tho.

Reply:
You can also use a before_update callback.